From bbceaac1b387e7bff68d788aae3a1486cda8d06a Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Wed, 31 May 2017 11:50:07 +0200
Subject: [PATCH 01/12] Move mfd/ drivers up in Makefile

CRC PMIC driver was loaded after the i915 GPU driver. However, the
i915 depends on the PWM and panel GPIO registered by the PMIC driver.

Consequently, it failed to load them, breaking backlight control.
Loading the PMIC driver before i915 fixes it.
---
 drivers/Makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/Makefile b/drivers/Makefile
index d242d3514d30..01b8ca04faef 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -61,6 +61,10 @@ obj-y				+= char/
 # iommu/ comes before gpu as gpu are using iommu controllers
 obj-$(CONFIG_IOMMU_SUPPORT)	+= iommu/
 
+# mfd/ comes before gpu/ because otherwise the backlight PWM for CRC is
+# registered too late
+obj-y				+= mfd/
+
 # gpu/ comes after char for AGP vs DRM startup and after iommu
 obj-y				+= gpu/
 
@@ -72,7 +76,7 @@ obj-$(CONFIG_FB_INTEL)          += video/fbdev/intelfb/
 
 obj-$(CONFIG_PARPORT)		+= parport/
 obj-$(CONFIG_NVM)		+= lightnvm/
-obj-y				+= base/ block/ misc/ mfd/ nfc/
+obj-y				+= base/ block/ misc/ nfc/
 obj-$(CONFIG_LIBNVDIMM)		+= nvdimm/
 obj-$(CONFIG_DAX)		+= dax/
 obj-$(CONFIG_DMA_SHARED_BUFFER) += dma-buf/
-- 
2.15.1

